pkg/interactor/save_data_test.go

```diff
@@ -0,0 +1,81 @@
+package interactor
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/CESARBR/knot-cloud-storage/pkg/entities"
+	"github.com/CESARBR/knot-cloud-storage/pkg/mocks"
+	"github.com/stretchr/testify/assert"
+)
+
+var date = time.Date(2020, time.April, 15, 0, 0, 0, 0, time.UTC)
+
+type SaveDataTestCase struct {
+	name             string
+	authorization    string
+	deviceID         string
+	payload          []entities.Payload
+	data             entities.Data
+	fakeLogger       *mocks.FakeLogger
+	fakeDataStore    *mocks.FakeDataStore
+	fakeThingService *mocks.FakeThingService
+}
+
+var sdCases = []SaveDataTestCase{
+	{
+		"authorization key not provided",
+		"",
+		"",
+		nil,
+		entities.Data{},
+		&mocks.FakeLogger{},
+		&mocks.FakeDataStore{},
+		&mocks.FakeThingService{ReturnErr: errors.New("no access token provided")},
+	},
+	{
+		"data successfully saved on the database",
+		"authorization-key",
+		"fc3fcf912d0c290a",
+		[]entities.Payload{{SensorID: 1, Value: 999}},
+		entities.Data{From: "fc3fcf912d0c290a", Payload: entities.Payload{SensorID: 1, Value: 999}, Timestamp: date},
+		&mocks.FakeLogger{},
+		&mocks.FakeDataStore{},
+		&mocks.FakeThingService{},
+	},
+	{
+		"failed to save the data on the database",
+		"authorization-key",
+		"fc3fcf912d0c290a",
+		[]entities.Payload{{SensorID: 1, Value: 999}},
+		entities.Data{From: "fc3fcf912d0c290a", Payload: entities.Payload{SensorID: 1, Value: 999}, Timestamp: date},
+		&mocks.FakeLogger{},
+		&mocks.FakeDataStore{},
```

Comment:
Shouldn't we return an error here to emulate the save failure?

Reply:
Done!